platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java

```diff
@@ -90,15 +96,77 @@ public static String addressBookConfigText(@NonNull final AddressBook addressBoo
     public static AddressBook parseAddressBookText(@NonNull final String addressBookText) throws ParseException {
         Objects.requireNonNull(addressBookText, "The addressBookText must not be null.");
         final AddressBook addressBook = new AddressBook();
-        for (final String addressLine : addressBookText.split("\\r?\\n")) {
-            final Address address = parseAddressText(addressLine);
-            if (address != null) {
-                addressBook.add(address);
+        boolean nextNodeIdParsed = false;
+        for (final String line : addressBookText.split("\\r?\\n")) {
+            final String trimmedLine = line.trim();
+            if (trimmedLine.isEmpty() || trimmedLine.startsWith("#")) {
+                continue;
             }
+            if (trimmedLine.startsWith(ADDRESS_KEYWORD)) {
+                final Address address = parseAddressText(trimmedLine);
+                if (address != null) {
+                    addressBook.add(address);
+                }
+            } else if (trimmedLine.startsWith(NEXT_NODE_ID_KEYWORD)) {
+                final NodeId nodeId = parseNextAvailableNodeId(trimmedLine);
+                addressBook.setNextNodeId(nodeId);
+                nextNodeIdParsed = true;
+            } else {
+                throw new ParseException(
+                        "The line [%s] does not start with `%s` or `%s`."
+                                .formatted(line.substring(0, 30), ADDRESS_KEYWORD, NEXT_NODE_ID_KEYWORD),
+                        0);
+            }
+        }
+        if (!nextNodeIdParsed) {
+            throw new ParseException("The address book text does not contain a `nextAvailableNodeId` line.", 0);
```

Comment:
You have several literal strings throughout this file that say "nextAvailableNodeId", which isn't the correct text

It's probably worth using your static string everywhere, so that things stay in sync with changes to the literal

Reply:
Agreed, making the change.